sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs

```diff
@@ -174,7 +177,7 @@ public SasQueryParameters ToSasQueryParameters(SharedKeyCredentials sharedKeyCre
 
             var signature = sharedKeyCredential.ComputeHMACSHA256(stringToSign);
 
-            var p = new SasQueryParameters(
+            var p = new BlobSasQueryParameters(
```

Comment:
NIT: Not a fan of `p` we could just do
```
return new BlobSasQueryParameters(
                version: this.Version, 
                services: null, 
                resourceTypes: null, 
                protocol: this.Protocol, 
                startTime: this.StartTime, 
                expiryTime: this.ExpiryTime, 
                ipRange: this.IPRange, 
                identifier: this.Identifier, 
                resource: this.Resource, 
                permissions: this.Permissions, 
                signature: signature);
```

Reply:
same in other places

Reply:
I generally tried not to touch existing code while making these changes.  I'm assuming somebody wanted this to aid while debugging?

Reply:
Maybe, but then.. `p` .. :|

I guess we can always clean up later but I will prefer for all the cleaning code to eventually be addressed